packages/hadron-document/src/element.ts

```diff
@@ -743,7 +749,13 @@ export class Element extends EventEmitter {
    * Collapses only the target element
    */
   collapse(): void {
-    if (!this._isExpandable(this.originalExpandableValue)) {
+    if (
+      !this._isExpandable(this.originalExpandableValue) &&
+      !(
+        this.isModified() &&
+        (this.currentType === 'Array' || this.currentType === 'Object')
+      )
```

Comment:
Feels like applying this change of logic to `_isExpandable` method itself would be a more logical place to make sure we can consistently check for the element that is expandable (this method is used in a few other places). Is there a reason we're doing it here instead?

Reply:
`_isExpandable` is a static method for checking a value passed into it, and used in places that aren't related to the element it's being called on. I think I would prefer if we move that method off of `element.ts` and call it something like `isValueExpandable`. And then in `element.ts` we also add a new `isExpandable` which has this logic in it. How does that sound?

Reply:
Yeah, moving it to a static method makes sense! I would say though that this still seems a bit curious to me that we will be using different approaches for checking expandable state (like if we're using type here, why the _isExpandable method not using types and looking at values directly, this seems like it might potentially lead to inconsistencies), but restructuring this as you are suggeting will already be a nice improvement!